kong/templates/nginx_kong.lua

```diff
@@ -103,7 +103,7 @@ server {
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Proto $scheme;
-        proxy_set_header Host $upstream_host;
+        proxy_set_header Host $upstream_host:$upstream_port;
```

Comment:
According to the HTTP specs, the port really is part of the Host value if it is not the default port being requested for a service. This port could very well be included in the `$upstream_host` variable on the Lua land, to avoid dispersing that logic. Besides and to highlight that fact, this new variable is currently undeclared.